src/mongocxx/collection.cpp

```diff
@@ -18,6 +18,7 @@
 #include <cstdint>
 #include <limits>
 #include <tuple>
+#include <unordered_set>
```

Comment:
My other idea for keeping the temporary view_or_values alive was to throw them in a set.  Would that be better?